source/extensions/filters/http/header_to_metadata/header_to_metadata_filter.cc

```diff
@@ -50,7 +57,18 @@ bool Config::configToVector(const ProtobufRepeatedRule& proto_rules,
       throw EnvoyException(error);
     }
 
-    vector.push_back(rule);
+    // Ensure value and regex_value_rewrite are not mixed.
+    // TODO(rgs1): remove this once we are on v4 and these fields are folded into a oneof.
+    if (!entry.on_header_present().value().empty() &&
+        entry.on_header_present().has_regex_value_rewrite()) {
+      throw EnvoyException("Cannot specificy both value and regex_value_rewrite");
+    }
+
+    if (entry.has_on_header_missing() && entry.on_header_missing().value().empty()) {
+      throw EnvoyException("Cannot specificy on_header_missing rule with an empty value");
```

Comment:
Probably want to explicitly note this change in behavior in the release notes.

nit: specify